Dockerfile

```diff
@@ -17,22 +17,19 @@ WORKDIR /app
 
 RUN apt-get update && apt-get install -y gcc build-essential libpq-dev musl-dev python3-dev
 
-# Set up venv
-RUN python -m venv /opt/venv
-# Make sure we use the virtualenv:
-ENV PATH="/opt/venv/bin:$PATH"
+ENV POETRY_VERSION=1.5.0
 
-COPY api/requirements.txt .
-
-# Make sure we are running latest pip and setuptools to avoid potential security warnings
 RUN pip install --upgrade pip
 RUN pip install --upgrade setuptools
+RUN pip install "poetry==$POETRY_VERSION"
 
-# Install our python dependencies
-RUN pip install -r requirements.txt
+# Set up venv
+RUN python -m venv /opt/venv
 
+COPY api/pyproject.toml api/poetry.lock ./
+RUN . /opt/venv/bin/activate && poetry install --only main
```

Comment:
Do we need multi-stage builds? Something like the following should work perfectly. We only have to cache the package installation step as suggested here - https://python-poetry.org/docs/faq/#poetry-busts-my-docker-cache-because-it-requires-me-to-copy-my-source-files-in-before-installing-3rd-party-dependencies

```
FROM python
COPY pyproject.toml poetry.lock .
RUN pip install poetry && poetry install --no-root --no-directory
COPY src/ ./src
RUN poetry install --no-dev
```

Reply:
Thanks for the suggestion. Implemented in new commits.